src/bindings.c

```diff
@@ -98,7 +106,37 @@ Binding *configure_binding(const char *bindtype, const char *modifiers, const ch
         ELOG("Keybinding has more than one Group specified, but your X server is always in precisely one group. The keybinding can never trigger.\n");
 
     struct Mode *mode = mode_from_name(modename, pango_markup);
-    TAILQ_INSERT_TAIL(mode->bindings, new_binding, bindings);
+
+    /* --remove specified without command: remove the existing key binding (if
+     * any) without replacement. */
+    if (remove != NULL && command == NULL) {
+        Binding *bind;
+        TAILQ_FOREACH (bind, mode->bindings, bindings) {
+            if (binding_same_key(bind, new_binding)) {
+                TAILQ_REMOVE(mode->bindings, bind, bindings);
+                return NULL;
+            }
+        }
+
+        return NULL;
+    }
+
+    /* --remove specified with command: find existing key binding and replace
+     * it, or append if no existing key binding found. */
+    bool found = false;
+    if (remove != NULL) {
+        Binding *bind;
+        TAILQ_FOREACH (bind, mode->bindings, bindings) {
+            if (binding_same_key(bind, new_binding)) {
+                TAILQ_REPLACE(mode->bindings, bind, new_binding, bindings);
+                found = true;
+                break;
+            }
+        }
+    }
+    if (!found) {
+        TAILQ_INSERT_TAIL(mode->bindings, new_binding, bindings);
+    }
```

Comment:
Not necessarily a strictly better way to write this but WDYT about this:

```suggestion

if (remove) {
    bool found = false;
    Binding *bind;
    TAILQ_FOREACH (bind, mode->bindings, bindings) {
        if (binding_same_key(bind, new_binding)) {
            if (command){
                /* --remove specified with command: find existing key binding
                 * and replace it, or append if no existing key binding found. */
                TAILQ_REPLACE(mode->bindings, bind, new_binding, bindings);
                found = true;
                break;
            } else {
                /* --remove specified without command: remove the existing key
                 * binding (if any) without replacement. */
                TAILQ_REMOVE(mode->bindings, bind, bindings);
                return NULL;
            }
        }
    }

    if (!command){
        return NULL;
    } else if (!found){
        TAILQ_INSERT_TAIL(mode->bindings, new_binding, bindings);
    }
}
```

Reply:
Thanks for the suggestion.

I think I prefer the current version because of its early return pattern, which keeps indentation and code path complexity at a minimum.